addons/sale/models/sale_order_line.py

```diff
@@ -1076,12 +1070,15 @@ def _prepare_invoice_line(self, **optional_values):
             'discount': self.discount,
             'price_unit': self.price_unit,
             'tax_ids': [Command.set(self.tax_id.ids)],
-            'analytic_tag_ids': [Command.set(self.analytic_tag_ids.ids)],
+            'analytic_distribution': self.analytic_distribution,
             'sale_line_ids': [Command.link(self.id)],
             'is_downpayment': self.is_downpayment,
         }
         if self.order_id.analytic_account_id:
-            res['analytic_account_id'] = self.order_id.analytic_account_id.id
+            if self.analytic_distribution:
+                res['analytic_distribution'] = {self.order_id.analytic_account_id.id: self.analytic_distribution.get(self.order_id.analytic_account_id.id, 0) + 100}
+            else:
+                res['analytic_distribution'] = {self.order_id.analytic_account_id.id: 100}
```

Comment:
> Not sure about this, but if we want something like that:
> 
> How does it work in the widget of the SO with the mandatory plans if the plan is set on the SO? 
It should not block anything, as it is an automatic flow, so even if we have a percentage problem with mandatory plans, it should work.

> Do we want to keep the separate field on the SO instead of on the SOL?
I guess so, as if not, you're changing the fact that any SO can be only fully part of a project or not part of at all. When you create a SO (without SOL yet), you would not know that it is part of a project...
It would also make things weird: will the default on the SOL override this account? What if you save the distribution, do you include the project account?